### PR TITLE
[DO NOT MERGE YET] [Time Services] Cleaned up language, organization, and formatting for Time services

### PR DESCRIPTION
**Pull request by kodster28**
Addresses PCX-2485.

Tried to streamline the docs a bit, offload some of the background info to the blog, and otherwise be a bit more consistent about code formatting and headers.

### CODE REVIEW

**Comment by kodster28** [comment]
Don't think I omitted anything that was required... but might be worth a once over by @lukevalenta or someone on the Research Eng team.

File: content/time-services/ntp/_index.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ weight: 2
...
+{{<render file="_ntp-definition.md">}}
 
-NTP works by having a client send a query packet out to an NTP server that then responds with its clock time. The client then computes an estimate of the difference between its clock and the remote clock and attempts to compensate for network delay in this. NTP client queries multiple servers and implements algorithms to select the best estimate, and rejects clearly wrong answers.
+## Background
+
+NTP works by having a client send a query packet out to an NTP server that then responds with its clock time. The client then computes an estimate of the difference between its clock and the remote clock and attempts to compensate for any network delay. The NTP client then queries multiple servers and implements algorithms to select the best estimate.
+
+Cloudflare does not implement leap smearing: NTP includes a Leap Indicator field [spec](https://tools.ietf.org/html/rfc5905#section-7.3) and the kernel will apply the leap second correction at the appropriate time. This is the behavior servers in pool.ntp.org share. Using servers that smear time along with servers that do not may lead to unpredictable and anomalous results.
```
Comment (marciocloudflare) [inline content/time-services/ntp/_index.md]:
```suggestion
Cloudflare does not implement leap smearing: NTP includes a Leap Indicator field [spec](https://tools.ietf.org/html/rfc5905#section-7.3) and the kernel will apply the leap second correction at the appropriate time. This is the behavior servers in `pool.ntp.org` share. Using servers that smear time along with servers that do not may lead to unpredictable and anomalous results.
```

File: content/time-services/ntp/usage.md [hunk 2]
```diff
@@ -8,35 +8,32 @@ meta:
...
+3.  Click **Date and Time**.
+4.  Go to the **Internet Time** tab.
+5.  Click **Change settings..**
+6.  For **Server:**, type `time.cloudflare.com` and click **Update now**.
+7.  Click **OK**.
 
-You should receive the following message, letting you know that you have successfully synchronized your time.
-
-![](/time-services/static/windowtime2.png)
+![Screenshot of updating the Date and Time settings on machine running Windows](/time-services/static/window.png)
```
Comment (marciocloudflare) [inline content/time-services/ntp/usage.md:39]: This screenshot seems ancient. Maybe we can delete it?

File: content/time-services/nts.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+2. **NTP authentication**: Authenticates NTP time syncronization packets using the results of the TLS handshake. For more information, refer to [RFC 8915](https://tools.ietf.org/html/rfc8915).
+
+## Next steps
+
+[Chrony](https://chrony.tuxfamily.org/doc/devel/chrony.conf.html) and [NTPSec](https://www.ntpsec.org/) have support for NTS. Read the relevant documentation for guidance on setting them up to point to our time service, `time.cloudflare.com`.
+
+If you would like to hear about the development of additional clients or
+updates on our service or would like to announce that your client
+supports NTS, email `time-services@cloudflare.com` to be added
+to our distribution list.
```
Comment (marciocloudflare) [inline content/time-services/nts.md]:
```suggestion
If you would like to hear about the development of additional clients or updates on our service or would like to announce that your client supports NTS, email `time-services@cloudflare.com` to be added to our distribution list.
```

Comment (marciocloudflare) [reply content/time-services/nts.md]: Deleted the paragraphs forcing weird sentence breaks.

File: content/time-services/roughtime/_index.md [hunk 4]
```diff
@@ -6,23 +6,18 @@ weight: 4
...
-Cloudflare Roughtime.
+[Roughtime](https://roughtime.googlesource.com/roughtime) is a simple, flexible, and secure authenticated time protocol developed by Google. 
+
+## Background
+
+Endpoints on the Internet often synchronize their clocks using the [Network Time Protocol (NTP)](/time-services/ntp/). NTP provides precise synchronization, but is frequently deployed without a means of authentication. This is due to a [combination of issues](https://www.usenix.org/conference/usenixsecurity16/technical-sessions/presentation/dowling).
+
+As a result, a man-in-the-middle attacker can easily influence a victim’s clock. By moving them back in time, the attacker can, for example, force a victim to accept an expired (and possibly compromised) TLS certificate or session ticket.
+
+For many applications, *precise* network time is not essential. It's sufficient to have *accurate* time to mitigate these kinds of attacks, such as within 10 seconds of real time. This observation is the primary motivation behind Roughtime.
```
Comment (marciocloudflare) [inline content/time-services/roughtime/_index.md]:
```suggestion
For many applications, *precise* network time is not essential. It is sufficient to have *accurate* time to mitigate these kinds of attacks, such as within 10 seconds of real time. This observation is the primary motivation behind Roughtime.
```

File: content/time-services/roughtime/recipes.md [hunk 5]
```diff
@@ -1,163 +1,158 @@
...
-used to sign the response. The delegation serves the same function as a
-traditional [X.509 certificate](https://en.wikipedia.org/wiki/X.509) on the web:
-the client first uses the root public key to verify the delegation, then uses
-the online public key to verify the response.
-
-ng response: 2018-09-12 16:59:39.141 -0700 PDT ±1s (in 26ms)
-_auditable_ in this manner makes them accountable to provide accurate time.
+generates an _online_ public/secret key pair. The root secret key is used to
+create a _delegation_ for the online public key and the online secret key is
+used to sign the response.
```
Comment (marciocloudflare) [inline content/time-services/roughtime/recipes.md]:
```suggestion
generates an _online_ public/secret key pair. The root secret key is used to create a _delegation_ for the online public key and the online secret key is used to sign the response. 
```

File: content/time-services/roughtime/recipes.md [hunk 6]
```diff
@@ -1,163 +1,158 @@
...
+Because the response is _auditable_, the protocol makes each client accountable to provide accurate time.
+
 The configuration also encodes the type of signature algorithm used by the
-server. (Currently only [Ed25519](https://en.wikipedia.org/wiki/EdDSA) is
-supported.) Lastly, the configuration contains a list of addresses where the
-service can be reached, and which transport protocol to use to reach them.
-(Currently only UDP is supported.)
+server (currently only [Ed25519](https://en.wikipedia.org/wiki/EdDSA) is
+supported). Lastly, the configuration contains a list of addresses where the
+service can be reached and which transport protocol to use to reach them (currently only UDP is supported).
```
Comment (marciocloudflare) [inline content/time-services/roughtime/recipes.md]:
```suggestion
server (currently only [Ed25519](https://en.wikipedia.org/wiki/EdDSA) is supported). Lastly, the configuration contains a list of addresses where the service can be reached and which transport protocol to use to reach them (currently only UDP is supported).
```

Comment (marciocloudflare) [reply content/time-services/roughtime/recipes.md]: Maybe there's a reason to have these weird sentence breaks? If so, please ignore these suggestions

File: content/time-services/roughtime/usage.md [hunk 7]
```diff
@@ -1,21 +1,39 @@
...
-change in the future. We will keep this page up-to-date with the most current
-public key. You can also obtain it programmatically using DNS. For example:
+You just need the server's address and public key to run the protocol:
 
-    $ dig TXT roughtime.cloudflare.com | grep -oP 'TXT\s"\K.*?(?=")'
-    gD63hSj3ScS+wuOeGrubXlq35N1c5Lby/S+T7MNTjxo=
+- **Server address**: `roughtime.cloudflare.com:2002` (resolves to an IP address in our [anycast IP range](https://www.cloudflare.com/learning/cdn/glossary/anycast-network/)). You can use either IPv4 or IPv6.
+- **Public key**: `gD63hSj3ScS+wuOeGrubXlq35N1c5Lby/S+T7MNTjxo=`
+
+To get started, download and run our Go client:
```
Comment (marciocloudflare) [inline content/time-services/roughtime/usage.md]:
```suggestion
To get started, download and run Cloudflare's [Go client](https://github.com/cloudflare/roughtime):
```

Comment (marciocloudflare) [reply content/time-services/roughtime/usage.md]: Is this the correct link?

File: content/time-services/roughtime/usage.md [hunk 8]
```diff
@@ -1,21 +1,39 @@
...
+
+You can also obtain it programmatically using DNS. For example:
+
+```sh
+$ dig TXT roughtime.cloudflare.com | grep -oP 'TXT\s"\K.*?(?=")'
+```
+
+## Next steps
+
+Beyond just getting the roughtime from Cloudflare, you may want to use it to [keep your clock in sync](/time-services/roughtime/recipes/).
```
Comment (marciocloudflare) [inline content/time-services/roughtime/usage.md]:
```suggestion
Beyond just getting the Roughtime from Cloudflare, you may want to use it to [keep your clock in sync](/time-services/roughtime/recipes/).
```

**Review by marciocloudflare — APPROVED** [review]

**Comment by kodster28** [comment]
Note to self, need to set up redirects for ntp user guide + roughtime protocol pages after merging.

File: content/time-services/nts.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+## Background
+
+The NTS protocol is divided into two phases:
+
+1. **NTS key exchange**: Establishes the necessary key material between the NTP client and the server, using a [Transport Layer Security (TLS) handshake](https://www.cloudflare.com/learning/ssl/what-happens-in-a-tls-handshake/) (the same public key infrastructure as the web). Once the keys are exchanged, the TLS channel is closed and the protocol enters the second phase. 
+2. **NTP authentication**: Authenticates NTP time syncronization packets using the results of the TLS handshake. For more information, refer to [RFC 8915](https://tools.ietf.org/html/rfc8915).
+
+## Next steps
+
+[Chrony](https://chrony.tuxfamily.org/doc/devel/chrony.conf.html) and [NTPSec](https://www.ntpsec.org/) have support for NTS. Read the relevant documentation for guidance on setting them up to point to our time service, `time.cloudflare.com`.
```
Comment (lukevalenta) [inline content/time-services/nts.md]: Looks like `NTPsec` is the capitalization used at https://www.ntpsec.org/.

```suggestion
[Chrony](https://chrony.tuxfamily.org/doc/devel/chrony.conf.html) and [NTPsec](https://www.ntpsec.org/) have support for NTS. Read the relevant documentation for guidance on setting them up to point to our time service, `time.cloudflare.com`.
```

File: content/time-services/roughtime/_index.md [hunk 10]
```diff
@@ -6,23 +6,18 @@ weight: 4
...
-    $ go install github.com/cloudflare/roughtime...
-    $ getroughtime -ping roughtime.cloudflare.com:2002 -pubkey gD63hSj3ScS+wuOeGrubXlq35N1c5Lby/S+T7MNTjxo=
-    ping response: 2018-09-12 16:59:39.141 -0700 PDT ±1s (in 10ms)
-
-That's it — authenticated time! Why is this interesting and what is it about
-the protocol that makes it "rough"?  Proceed to learn more about Roughtime and
-its security features with a more in-depth look at our
-[blog post](https://blog.cloudflare.com/roughtime/) about the launch of
-Cloudflare Roughtime.
+[Roughtime](https://roughtime.googlesource.com/roughtime) is a simple, flexible, and secure authenticated time protocol developed by Google.
```
Comment (lukevalenta) [inline content/time-services/roughtime/_index.md:9]: @cjpatton think we should mention here that our implementation is not currently compatible with the IETF draft? https://datatracker.ietf.org/doc/html/draft-ietf-ntp-roughtime

**Review by lukevalenta — APPROVED** [review]
The cleanup looks great--thank you @kodster28!

**Comment by Lar1o87** [comment]
ບໍ່ຕ້ອງສົງມາກະໄດ້ມີຫາຍແລ້ວ

ເມື່ອ ພຫ., 24 ມ.ນ. 2022, 21:16 Luke Valenta ***@***.***>
ຂຽນວ່າ:

File: content/time-services/roughtime/_index.md (hunk 10 above)
Comment (lukevalenta) [reply content/time-services/roughtime/_index.md:9]: Fwiw, this isn't a blocker to merging this cleanup PR

File: content/time-services/ntp/usage.md (hunk 2 above)
Comment (kodster28) [reply content/time-services/ntp/usage.md:39]: I'll keep for now, but thanks for the callout!

**Pull request merged**